provisioner/inspec/provisioner.go

```diff
@@ -0,0 +1,563 @@
+package inspec
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+	"os/user"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"unicode"
+
+	"golang.org/x/crypto/ssh"
+
+	"github.com/hashicorp/packer/common"
+	"github.com/hashicorp/packer/common/adapter"
+	"github.com/hashicorp/packer/helper/config"
+	"github.com/hashicorp/packer/packer"
+	"github.com/hashicorp/packer/template/interpolate"
+)
+
+var SupportedBackends = map[string]bool{"docker": true, "local": true, "ssh": true, "winrm": true}
+
+type Config struct {
+	common.PackerConfig `mapstructure:",squash"`
+	ctx                 interpolate.Context
+
+	// The command to run inspec
+	Command    string
+	SubCommand string
+
+	// Extra options to pass to the inspec command
+	ExtraArguments []string `mapstructure:"extra_arguments"`
+	InspecEnvVars  []string `mapstructure:"inspec_env_vars"`
+
+	// The profile to execute.
+	Profile              string   `mapstructure:"profile"`
+	AttributesDirectory  string   `mapstructure:"attributes_directory"`
+	AttributesFiles      []string `mapstructure:"attributes"`
+	Backend              string   `mapstructure:"backend"`
+	User                 string   `mapstructure:"user"`
+	Host                 string   `mapstructure:"host"`
+	LocalPort            uint     `mapstructure:"local_port"`
+	SSHHostKeyFile       string   `mapstructure:"ssh_host_key_file"`
+	SSHAuthorizedKeyFile string   `mapstructure:"ssh_authorized_key_file"`
+}
+
+type Provisioner struct {
+	config           Config
+	adapter          *adapter.Adapter
+	done             chan struct{}
+	inspecVersion    string
+	inspecMajVersion uint
+}
+
+func (p *Provisioner) Prepare(raws ...interface{}) error {
```

Comment:
It might be nice to validate in the Prepare() that inspec is in Packer's path so that folks with pathing issues don't wait to find that out until the run fails -- especially since inspec will probably be the last provisioner run.